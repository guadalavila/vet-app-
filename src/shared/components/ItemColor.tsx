import React from 'react';
import { StyleSheet, View } from 'react-native';

interface IItemColorProps {
    size?: number;
    color: string;
}

const ItemColor = ({ size = 30, color }: IItemColorProps) => {
    return (
        <View
            testID='ItemColor'
            style={[styles.container, { width: size, height: size, borderRadius: size, backgroundColor: color }]}
        />
    );
};

export default ItemColor;

const styles = StyleSheet.create({
    container: {},
});
