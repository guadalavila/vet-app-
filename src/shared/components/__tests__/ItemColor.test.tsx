import { render, screen } from '@testing-library/react-native';
import React from 'react';
import ItemColor from '../ItemColor';

describe('ItemColor', () => {
    test('should match snapshot', () => {
        const container = render(<ItemColor size={33} color='black' />);
        expect(container).toMatchSnapshot();
    });

    test('should render component', () => {
        render(<ItemColor size={33} color='black' />);
        expect(screen.getByTestId('ItemColor')).toBeTruthy();
    });

    test('should render correct styles', () => {
        render(<ItemColor size={33} color='black' />);
        expect(screen.getByTestId('ItemColor').props.style).toEqual([
            {},
            { backgroundColor: 'black', borderRadius: 33, height: 33, width: 33 },
        ]);
    });
});
