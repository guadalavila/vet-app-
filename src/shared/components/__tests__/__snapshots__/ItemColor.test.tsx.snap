// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemColor should match snapshot 1`] = `
<View
  style={
    Array [
      Object {},
      Object {
        "backgroundColor": "black",
        "borderRadius": 33,
        "height": 33,
        "width": 33,
      },
    ]
  }
  testID="ItemColor"
/>
`;
